compiler.go

```diff
@@ -110,7 +110,16 @@ func NewCompiler() *Compiler {
 //
 // Note that url must not have fragment
 func (c *Compiler) AddResource(url string, r io.Reader) error {
-	res, err := newResource(url, r)
+	doc, err := unmarshal(r)
+	if err != nil {
+		return fmt.Errorf("jsonschema: invalid json %s: %v", url, err)
+	}
+	return c.AddResourceJSON(url, doc)
+}
+
+// AddResourceJSON adds a schema from an already loaded map to the compiler.
```

Comment:
change comment: `AddResourceJSON adds in-memory resource from given json value.`

Reply:
applied